### PR TITLE
feature/8605-chris-appointments-community-care

### PR DESCRIPTION
**Pull request by alexandec**
## Description of Change

Add CommunityCareAppointment component, update shared components, update unit tests.

## Screenshots/Video

Upcoming Community Care
&nbsp;&nbsp;

Past Community Care
&nbsp;&nbsp;

Canceled Community Care
&nbsp;&nbsp;

Pending Community Care
&nbsp;&nbsp;

Canceled Pending Community Care
&nbsp;&nbsp;

## Testing

- [ ] Tested on iOS 
- [ ] Tested on Android 

## Reviewer Validations

Reference [the spreadsheet](https://docs.google.com/spreadsheets/d/1Q_kdxH1zJjmJ0xF5XC7aMUVzlwxtO3wAEvYxFrTACsc/edit#gid=402791918), specifically Community Care

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/ClaimExamAppointment.test.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ import ClaimExamAppointment from './ClaimExamAppointment'
 context('ClaimExamAppointment', () => {
   const defaultAppointmentAttributes: AppointmentAttributes = {
     //appointmentType and Status not used at this point in the logic, those are used in the upcoming appointments details
-    appointmentType: AppointmentTypeConstants.VA,
+    appointmentType: AppointmentTypeConstants.COMMUNITY_CARE,
```
Comment (Sparowhawk) [inline VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/ClaimExamAppointment.test.tsx:19]: does this one need to be changed?

**Review by Sparowhawk — COMMENTED** [review]
Shouldn't the CommunityCarePendingLocation file be in the shared components folder and that index file be updated accordingly?

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/ClaimExamAppointment.test.tsx (hunk 1 above)
Comment (alexandec) [reply VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/ClaimExamAppointment.test.tsx]: Fixed, thanks

**Comment by alexandec** [comment]
I didn't put CommunityCarePendingLocation in the SharedComponents directory since it's not shared. It's only used by the CommunityCareAppointment component. I'm open to putting it in SharedComponents if you think that makes sense though

**Comment by Sparowhawk** [comment]
> I didn't put CommunityCarePendingLocation in the SharedComponents directory since it's not shared. It's only used by the CommunityCareAppointment component. I'm open to putting it in SharedComponents if you think that makes sense though

I think we should. I am putting the join session/prepare link as a shared component in the event this ever gets used by another appt type. I think it makes sense that this component could be reused in the future and we should probably treat it as such.

**Comment by alexandec** [comment]
Sure, moved that component

**Review by Sparowhawk — APPROVED** [review]

File: VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentLocation.tsx [hunk 2]
```diff
@@ -43,12 +44,26 @@ const locationHeading = (subType: AppointmentDetailsSubType, type: AppointmentDe
...
   }
 }
 
+const getCommunityCareProvider = (attributes: AppointmentAttributes, type: AppointmentDetailsScreenType) => {
+  const { healthcareProvider } = attributes
+
+  if (type === AppointmentDetailsTypeConstants.CommunityCare && healthcareProvider) {
+    return (
+      <TextView variant="MobileBody" selectable={true}>
+        {healthcareProvider}
```
Comment (alexandec) [inline VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentLocation.tsx:61]: For now I'm showing healthcareProvider here. Based on @rbontrager's findings we may need to request that the BE team add a different field to show instead, but we aren't sure yet

**Review by rbontrager — APPROVED** [review]

**Pull request merged**